### PR TITLE
util/kvencoder: add more unit test

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
add more unit test for util/kvencoder.

### What is changed and how it works?
```
...
OK: 14 passed
PASS
coverage: 86.8% of statements
ok      github.com/pingcap/tidb/util/kvencoder  1.835s
```

### Check List 

Tests 

 - No code

### CODE REVIEW

File: util/kvencoder/kv_encoder_test.go [hunk 1]
```diff
@@ -438,6 +438,35 @@ func (s *testKvEncoderSuite) TestAllocatorRebase(c *C) {
...
+
+func (s *testKvEncoderSuite) TestError(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	alloc := NewAllocator()
+	encoder, err := New("test", alloc)
+	c.Assert(err, IsNil)
+	defer encoder.Close()
+	_, err = New("", alloc)
+	c.Assert(err, NotNil)
```
Comment (xiekeyi98) [inline util/kvencoder/kv_encoder_test.go]: I think to use `c.Assert( err, ErrorMatches , regex)` is better than this.
May we could test the error message better.

Comment (bb7133) [reply util/kvencoder/kv_encoder_test.go]: Agree

Comment (crazycs520) [reply util/kvencoder/kv_encoder_test.go]: done.

**Pull request merged**